packages/kit/src/vite/index.js

```diff
@@ -439,14 +442,35 @@ function collect_output(bundle) {
  * @param {Record<string, any>} config
  * @param {Record<string, any>} resolved_config
  */
-function warn_overridden_config(config, resolved_config) {
+function create_overridden_config_action(config, resolved_config) {
 	const overridden = find_overridden_config(config, resolved_config, enforced_config, '', []);
 	if (overridden.length > 0) {
-		console.log(
-			colors.bold().red('The following Vite config options will be overridden by SvelteKit:')
-		);
-		console.log(overridden.map((key) => `  - ${key}`).join('\n'));
+		return () => {
+			console.log(
+				colors.bold().red('The following Vite config options will be overridden by SvelteKit:')
+			);
+			console.log(overridden.map((key) => `  - ${key}`).join('\n'));
+		};
 	}
+	return () => {};
+}
+
+/**
+ * @param {import('vite').ViteDevServer} server
+ * @param {Array<() => void>} actions
+ */
+function add_post_listening_actions(server, ...actions) {
+	const _listen = server.listen;
+	server.listen = function () {
+		server.httpServer?.on('listening', () => {
+			setTimeout(() => {
+				actions.forEach((action) => action());
+			}, 100);
+		});
+		// @ts-ignore
+		// eslint-disable-next-line prefer-rest-params
+		return _listen.apply(this, arguments);
```

Comment:
If we don't need to do anything special on restart, I think we could skip re-assigning server.listen, and attach the `listening` listener only 🤔  That way we can set this up for the `preview` too by using `configurePreviewServer`.

Re 100ms, I saw your discord message, and I'm not sure why it's nor working reliably. 100ms is probably fine for now though.